src/software/ai/hl/stp/action/action.h

```diff
@@ -39,6 +39,11 @@ class Action
      */
     bool done() const;
 
+    /**
+     * runs the coroutine from the start, effectively restarting the action
```

Comment:
Hmm, I'd prefer not to mention "coroutine" in the interface because it's technically an implementation detail, but seems somewhat relevant here. Thoughts @MathewMacDougall @akhilveeraghanta @jonathanlew ?

Reply:
Agree that I'd prefer not to mention it here if possible. I think we can just reword it to refer to the state, since Actions are known to be stateful.

eg. something like
"Resets any state and restarts the Action from the beginning"

Do you think that sufficiently gets the message across?

Reply:
I think your suggestion here is good.

Reply:
Agreed, changed comment to Matt's suggestion